src/lib/api.js

```diff
@@ -229,15 +229,25 @@ export default class RocketChatInstance {
   }
 
   async realtime(callback) {
+    let subscriptionId;
     try {
       await this.rcClient.connect();
       await this.rcClient.resume({ token: Cookies.get(RC_USER_TOKEN_COOKIE) });
-      await this.rcClient.subscribe('stream-room-messages', this.rid);
-      await this.rcClient.onMessage((data) => {
+      subscriptionId = await this.rcClient.subscribe(
+        'stream-room-messages',
+        this.rid
+      );
+
+      const messageHandler = (data) => {
+        this.rcClient.unsubscribe(subscriptionId);
```

Comment:
If we are unsubscribing it, are we maintaining the realtime chat functionality?

Reply:
Yes i showed that in preview also real-time communication is working perfectly fine